specification/cat/allocation/types.ts

```diff
@@ -19,14 +19,31 @@
 
 import { ByteSize } from '@_types/common'
 import { Host, Ip } from '@_types/Networking'
-import { Percentage } from '@_types/Numeric'
+import { double, Percentage } from '@_types/Numeric'
 
 export class AllocationRecord {
   /**
    * Number of primary and replica shards assigned to the node.
    * @aliases s
    */
   shards?: string
+
+  /**
+   * Amount of shards that are scheduled to be moved elsewhere in the cluster or -1 other than desired balance allocator is used
+   */
+  'shards.undesired'?: string | null
```

Comment:
From the linked server-code it seems like these properties are always present. Should we remove the optional modifier?